packages/trader/src/Modules/Trading/Components/Form/ContractType/contract-type-widget.jsx

```diff
@@ -5,107 +5,95 @@ import { localize } from '@deriv/translations';
 import ContractType from './contract-type.jsx';
 import { getContractTypeCategoryIcons } from '../../../Helpers/contract-type';
 
-class ContractTypeWidget extends React.PureComponent {
-    state = {
-        is_dialog_open: false,
-        is_info_dialog_open: false,
-        list: this.props.list,
-        selected_category: null,
-        search_query: '',
-    };
-
-    componentDidMount() {
-        document.addEventListener('mousedown', this.handleClickOutside);
-    }
-
-    componentWillUnmount() {
-        document.removeEventListener('mousedown', this.handleClickOutside);
-    }
-
-    componentDidUpdate() {
-        if (this.props.list.length !== this.state.list.length && !this.state.on_input_change) {
-            // reset internal list state when contracts list is changed in store (on contracts_for response)
-            this.setState({ list: this.props.list });
+const ContractTypeWidget = ({ is_equal, name, value, list, onChange }) => {
+    const wrapper_ref = React.useRef(null);
+    const [is_dialog_open, setDialogVisibility] = React.useState(false);
+    const [is_info_dialog_open, setInfoDialogVisibility] = React.useState(false);
+    const [localList, setLocalList] = React.useState(list);
```

Comment:
Is this a list based on user's location? not sure if `localList` is a very descriptive name!

Reply:
it is an internal list state which should change when contracts list from store  is changed(on contracts_for response).
is internal_list better choice for it's name?